File: decision-agent/agent-backend/agent-factory/src/drivenadapter/httpaccess/v2agentexecutoraccess/v2agentexecutordto/interrupt_data.go
package v2agentexecutordto

// InterruptData 中断详情
type InterruptData struct {
	ToolName        string           `json:"tool_name"`        // 工具名称
	ToolDescription string           `json:"tool_description"` // 工具描述
	ToolArgs        []ToolArg        `json:"tool_args"`        // 工具参数列表
	InterruptConfig *InterruptConfig `json:"interrupt_config"` // 中断配置
}

// ToolArg 工具参数
type ToolArg struct {
	Key   string      `json:"key"`   // 参数名称
	Value interface{} `json:"value"` // 参数值
	Type  string      `json:"type"`  // 参数类型
}

// InterruptConfig 中断配置
type InterruptConfig struct {
	RequiresConfirmation bool   `json:"requires_confirmation"` // 是否需要用户确认
	ConfirmationMessage  string `json:"confirmation_message"`  // 确认提示消息
}

File: decision-agent/agent-backend/agent-executor/app/domain/vo/interrupt/interrupt_data.py
# -*- coding:utf-8 -*-
"""中断详情 VO"""

from typing import List, Any, Optional
from pydantic import BaseModel, Field


class ToolArg(BaseModel):
    """工具参数"""

    key: str = Field(..., description="参数名称")
    value: Any = Field(..., description="参数值")
    type: str = Field(..., description="参数类型")


class InterruptConfig(BaseModel):
    """中断配置"""

    requires_confirmation: bool = Field(..., description="是否需要用户确认")
    confirmation_message: str = Field(..., description="确认提示消息")


class InterruptData(BaseModel):
    """中断详情"""

    tool_name: str = Field(..., description="工具名称")
    tool_description: Optional[str] = Field(None, description="工具描述")
    tool_args: List[ToolArg] = Field(default_factory=list, description="工具参数列表")
    interrupt_config: Optional[InterruptConfig] = Field(None, description="中断配置")
